hypothesis-python/src/hypothesis/strategies/_internal/lazy.py

```diff
@@ -30,32 +31,33 @@
 def unwrap_strategies(s):
     global unwrap_depth
 
-    if not isinstance(s, SearchStrategy):
+    # optimization
+    if not isinstance(s, (LazyStrategy, DeferredStrategy)):
```

Comment:
previously `not (isinstance(s, SearchStrategy) and hasattr(s, "wrapped_strategy"))`, but the hasattr caused the `@property(wrapped_strategy)` to execute, which was fine normally but caused crosshair tracing-outside-context errors (for some reason).